src/common/rpc/server_client.rs

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2022 Vaticle
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use std::sync::Arc;
+
+use futures::channel::mpsc;
+use tonic::Streaming;
+use typedb_protocol::{core_database, core_database_manager, session, transaction};
+
+use crate::{
+    async_enum_dispatch,
+    common::{
+        rpc::{core_client::CoreClient, ClusterClient},
+        Executor, Result,
+    },
+};
+
+#[derive(Clone, Debug)]
+pub(crate) enum ServerClient {
+    Core(CoreClient),
+    Cluster(ClusterClient),
+}
```

Comment:
Used by things that only connect to a single server (core server / cluster node, e.g. `server::DatabaseManager` or `Transaction`). We need them to have access to `ClusterClient` for its token renewal mechanism to engage correctly.